third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -529,8 +538,123 @@ impl InterpreterImpl {
+                                return Err(set_err_info!(current_frame, err));
+                            };
+
+                            // Charge gas for function code loading.
+                            let arena_id = traversal_context
+                                .referenced_module_ids
+                                .alloc(module_id.clone());
+                            loader.check_dependencies_and_charge_gas(
+                                resolver.module_store(),
+                                data_store,
+                                gas_meter,
+                                &mut traversal_context.visited,
+                                traversal_context.referenced_modules,
+                                [(arena_id.address(), arena_id.name())],
+                                resolver.module_storage(),
+                            )?;
+
+                            // Charge gas for code loading of modules used by type arguments.
+                            let modules_used_by_ty_args = ty_arg_tags
+                                .iter()
+                                .flat_map(|ty_tag| ty_tag.preorder_traversal_iter())
+                                .filter_map(TypeTag::struct_tag)
+                                .map(|struct_tag| {
+                                    let module_id = traversal_context
+                                        .referenced_module_ids
+                                        .alloc(struct_tag.module_id());
+                                    (module_id.address(), module_id.name())
+                                })
+                                .collect::<BTreeSet<_>>();
+                            loader.check_dependencies_and_charge_gas(
+                                resolver.module_store(),
+                                data_store,
+                                gas_meter,
+                                &mut traversal_context.visited,
+                                traversal_context.referenced_modules,
+                                modules_used_by_ty_args,
+                                resolver.module_storage(),
+                            )?;
+
+                            Ok(module_id.clone())
+                        },
+                    )?;
+
+                    // Resolve the function. This may lead to loading the code related
+                    // to this function.
+                    let callee = lazy_function
+                        .with_resolved_function(module_storage, |f| Ok(f.clone()))
+                        .map_err(|e| set_err_info!(current_frame, e))?;
+
+                    // Charge gas for call and for the parameters
+                    gas_meter
+                        .charge_call(
+                            &module_id,
+                            callee.name(),
+                            self.operand_stack
+                                .last_n(callee.param_tys().len() - mask.captured_count() as usize)
```

Comment:
Why we do not charge for all parameters? In the end, you still copy captured args to locals, so gas for call should account for all?

Reply:
Done